02-04-domain-validations/internal/platform/server/handler/courses/search_all.go

```diff
@@ -0,0 +1,38 @@
+package courses
+
+import (
+	"net/http"
+
+	mooc "github.com/CodelyTV/go-hexagonal_http_api-course/02-04-domain-validations/internal"
+	"github.com/gin-gonic/gin"
+)
+
+type CourseResponse struct {
+	Id       string
+	Name     string
+	Duration string
+}
+
+// CreateHandler returns an HTTP handler for courses creation.
+func SearchAllHandler(courseRepository mooc.CourseRepository) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+
+		courses, err := courseRepository.SearchAll(ctx)
+		if err != nil {
+			ctx.JSON(http.StatusBadRequest, err.Error())
+			return
+		}
+
+		var coursesResponse []CourseResponse
```

Comment:
```suggestion
		coursesResponse := make([]CourseResponse, 0, len(courses))
```

You can pre-allocate the required memory to make that data mapping a bit more efficient.